browser/main/SideNav/StorageItem.js

```diff
@@ -131,8 +132,64 @@ class StorageItem extends React.Component {
     }
   }
 
+  handleDragEnter (e) {
+    e.dataTransfer.setData('defaultColor', e.target.style.backgroundColor)
+    e.target.style.backgroundColor = 'rgba(129, 130, 131, 0.08)'
+  }
+
+  handleDragLeave (e) {
+    e.target.style.opacity = '1'
+    e.target.style.backgroundColor = e.dataTransfer.getData('defaultColor')
+  }
+
+  handleDrop (e, storage, folder, dispatch, location) {
+    e.target.style.opacity = '1'
+    e.target.style.backgroundColor = e.dataTransfer.getData('defaultColor')
+    const noteData = JSON.parse(e.dataTransfer.getData('note'))
+    if (folder.key !== noteData.folder) {
+      dataApi
+       .createNote(storage.key, {
+         content: noteData.content,
+         createdAt: noteData.createdAt,
+         folder: folder.key,
+         isStarred: noteData.isStarred,
+         storage: storage,
+         title: noteData.title,
+         tags: noteData.tags,
+         type: noteData.type,
+         updatedAt: noteData.updatedAt,
+         description: noteData.description,
+         snippets: noteData.snippets
```

Comment:
`Object.assign()` is better in this situation. `const newNoteData = Object.assign({}, noteData, { storage: storage, folder: folder.key })`